src/TitleDetails/TitleDetailsContainer.js

```diff
@@ -64,16 +101,61 @@ const TitleDetailsContainer = ({ location, history, mutator, match }) => {
     [history, title.id, location.search],
   );
 
-  if (isLoading) {
+  const onAddPiece = useCallback(
+    (values) => {
+      const mutatorMethod = values.id ? 'PUT' : 'POST';
+
+      mutator.orderPieces[mutatorMethod](values)
+        .then(() => showCallout({
+          messageId: 'ui-receiving.piece.actions.addPiece.success',
+          type: 'success',
+          values: { title: title.title },
+        }))
+        .catch(() => showCallout({ messageId: 'ui-receiving.piece.actions.addPiece.error', type: 'error' }))
+        .finally(() => fetchPieces(poLine.id));
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [fetchPieces, title.title],
+  );
+
+  const onCheckIn = useCallback(
+    (values) => {
+      const mutatorMethod = values.id ? 'PUT' : 'POST';
+
+      mutator.orderPieces[mutatorMethod](values)
+        .then(piece => {
+          showCallout({
+            messageId: 'ui-receiving.piece.actions.addPiece.success',
+            type: 'success',
+            values: { title: title.title },
+          });
+
+          return checkInItems([{
+            ...piece,
+            itemStatus: ITEM_STATUS.inProcess,
+          }], mutator.checkIn);
+        })
+        .catch(() => showCallout({ messageId: 'ui-receiving.piece.actions.checkInItem.error', type: 'error' }))
+        .finally(() => fetchPieces(poLine.id));
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [fetchPieces, title.title],
+  );
+
+  if (isLoading || !locations || !pieces) {
```

Comment:
`pieces` is included in `isLoading` chain

Reply:
But in cases `onAddItem()` and `onCheckIn` it's not included